lib/ansible/modules/network/nxos/nxos_ospf_vrf.py

```diff
@@ -424,31 +361,27 @@ def main():
             elif str(value).lower() == 'false':
                 value = False
             elif str(value).lower() == 'default':
-                value = PARAM_TO_DEFAULT_KEYMAP.get(key)
+                value = PARAM_TO_DEFAULT_KEYMAP.key
                 if value is None:
                     value = 'default'
-            if existing.get(key) or (not existing.get(key) and value):
+            if existing.get(key) != value:
                 proposed[key] = value
 
-    result = {}
-    if state == 'present' or (state == 'absent' and existing):
-        candidate = CustomNetworkConfig(indent=3)
-        invoke('state_%s' % state, module, existing, proposed, candidate)
-        response = load_config(module, candidate)
-        result.update(response)
+    candidate = CustomNetworkConfig(indent=3)
+    if state == 'present':
+        state_present(module, existing, proposed, candidate)
+    if state == 'absent' and existing:
+        state_absent(module, existing, proposed, candidate)
 
-    else:
-        result['updates'] = []
-
-    if module._verbosity > 0:
-        end_state = invoke('get_existing', module, args)
-        result['end_state'] = end_state
-        result['existing'] = existing
-        result['proposed'] = proposed_args
+    if candidate:
+        load_config(module, candidate)
```

Comment:
nxapi will fail here when given a CustomNetworkConfig directly. In order to avoid this, call `.items_text()` before `load_config()` like so:

```
candidate = candidate.items_text()
load_config(module, candidate)
result['changed'] = True
result['commands'] = candidate
```